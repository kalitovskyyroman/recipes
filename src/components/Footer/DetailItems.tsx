import TwitterIcon from "@mui/icons-material/Twitter";
import YouTubeIcon from "@mui/icons-material/YouTube";
import FacebookIcon from "@mui/icons-material/Facebook";
import InstagramIcon from "@mui/icons-material/Instagram";
import PinterestIcon from "@mui/icons-material/Pinterest";

const socials = [
    {
        url: "some twitter url",
        logo: <TwitterIcon />,
    },
    {
        url: "some YouTube url",
        logo: <YouTubeIcon />,
    },
    {
        url: "some Facebook url",
        logo: <FacebookIcon />,
    },
    {
        url: "some Instagram url",
        logo: <InstagramIcon />,
    },
    {
        url: "some Pinterest url",
        logo: <PinterestIcon />,
    },
];

const basicLinks = [
    {
        title: "contact us",
    },
    {
        title: "about us",
    },
    {
        title: "subscribe",
    },
];

export { socials, basicLinks };
