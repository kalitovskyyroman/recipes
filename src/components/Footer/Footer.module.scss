@import "src/styles/mixins/breakpoints.module.scss";
@import "src/styles/colors.module.scss";
@import "src/styles/consts/sizes.module.scss";

.footer {
    color: white;
    background-color: $rc-black;
    padding: 30px $general-padding 20px;

    a:hover {
        color: $rc-orange;
    }

    .first-row {
        .title {
            text-align: left;
            margin-bottom: 15px;
            font-size: 40px;

            @include breakpoints($tablet) {
                margin-right: 10%;
            }
        }

        .socials {
            text-align: left;
            margin-bottom: 20px;

            a {
                margin-right: 20px;
            }

            @include breakpoints($tablet) {
                display: flex;
                align-items: center;
                flex-direction: row;
                margin-bottom: 0;
            }
        }

        @include breakpoints($tablet) {
            display: flex;
            justify-content: left;
        }
    }

    .pages {
        a {
            display: block;
            text-transform: capitalize;
            margin-bottom: 10px;

            .page-link-title {
                font-size: 20px;
            }
        }

        @include breakpoints($tablet) {
            display: flex;
            justify-content: space-between;
        }
    }

    @include breakpoints($tablet) {
        padding: 30px 100px;
    }

    @include breakpoints($desktop) {
        padding: 50px 150px;
    }
}
