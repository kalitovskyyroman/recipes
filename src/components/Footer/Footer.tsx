import { Typography } from "@mui/material";
import { socials, basicLinks } from "./DetailItems";

import styles from "./Footer.module.scss";

const Footer = () => (
    <div className={styles.footer}>
        <div className={styles["first-row"]}>
            <Typography className={styles.title} variant="h3">
                Recipes
            </Typography>
            <div className={styles.socials}>
                {socials.map(social => (
                    <a key={social.url} href={social.url} rel="noreferrer" target="_blank">
                        {social.logo}
                    </a>
                ))}
            </div>
        </div>
        <div className={styles.pages}>
            {basicLinks.map(page => (
                <a key={page.title} href="_#">
                    <Typography className={styles["page-link-title"]} variant="body1">
                        {page.title}
                    </Typography>
                </a>
            ))}
        </div>
    </div>
);

export default Footer;
