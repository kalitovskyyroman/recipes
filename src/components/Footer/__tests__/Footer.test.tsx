import TestRenderer from "react-test-renderer";
import Footer from "../Footer";

describe("Footer component", () => {
    test("Render footer", async () => {
        const component = TestRenderer.create(<Footer />);
        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();
    });
});
