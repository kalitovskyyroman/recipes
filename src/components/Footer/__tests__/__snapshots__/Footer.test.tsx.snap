// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component Render footer 1`] = `
<div
  className="footer"
>
  <div
    className="first-row"
  >
    <h3
      className="MuiTypography-root MuiTypography-h3 title css-gepadz-MuiTypography-root"
    >
      Recipes
    </h3>
    <div
      className="socials"
    >
      <a
        href="some twitter url"
        rel="noreferrer"
        target="_blank"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="TwitterIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M22.46 6c-.77.35-1.6.58-2.46.69.88-.53 1.56-1.37 1.88-2.38-.83.5-1.75.85-2.72 1.05C18.37 4.5 17.26 4 16 4c-2.35 0-4.27 1.92-4.27 4.29 0 .34.04.67.11.98C8.28 9.09 5.11 7.38 3 4.79c-.37.63-.58 1.37-.58 2.15 0 1.49.75 2.81 1.91 3.56-.71 0-1.37-.2-1.95-.5v.03c0 2.08 1.48 3.82 3.44 4.21a4.22 4.22 0 0 1-1.93.07 4.28 4.28 0 0 0 4 2.98 8.521 8.521 0 0 1-5.33 1.84c-.34 0-.68-.02-1.02-.06C3.44 20.29 5.7 21 8.12 21 16 21 20.33 14.46 20.33 8.79c0-.19 0-.37-.01-.56.84-.6 1.56-1.36 2.14-2.23z"
          />
        </svg>
      </a>
      <a
        href="some YouTube url"
        rel="noreferrer"
        target="_blank"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="YouTubeIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 15l5.19-3L10 9v6m11.56-7.83c.13.47.22 1.1.28 1.9.07.8.1 1.49.1 2.09L22 12c0 2.19-.16 3.8-.44 4.83-.25.9-.83 1.48-1.73 1.73-.47.13-1.33.22-2.65.28-1.3.07-2.49.1-3.59.1L12 19c-4.19 0-6.8-.16-7.83-.44-.9-.25-1.48-.83-1.73-1.73-.13-.47-.22-1.1-.28-1.9-.07-.8-.1-1.49-.1-2.09L2 12c0-2.19.16-3.8.44-4.83.25-.9.83-1.48 1.73-1.73.47-.13 1.33-.22 2.65-.28 1.3-.07 2.49-.1 3.59-.1L12 5c4.19 0 6.8.16 7.83.44.9.25 1.48.83 1.73 1.73z"
          />
        </svg>
      </a>
      <a
        href="some Facebook url"
        rel="noreferrer"
        target="_blank"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="FacebookIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M5 3h14a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2m13 2h-2.5A3.5 3.5 0 0 0 12 8.5V11h-2v3h2v7h3v-7h3v-3h-3V9a1 1 0 0 1 1-1h2V5z"
          />
        </svg>
      </a>
      <a
        href="some Instagram url"
        rel="noreferrer"
        target="_blank"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="InstagramIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7.8 2h8.4C19.4 2 22 4.6 22 7.8v8.4a5.8 5.8 0 0 1-5.8 5.8H7.8C4.6 22 2 19.4 2 16.2V7.8A5.8 5.8 0 0 1 7.8 2m-.2 2A3.6 3.6 0 0 0 4 7.6v8.8C4 18.39 5.61 20 7.6 20h8.8a3.6 3.6 0 0 0 3.6-3.6V7.6C20 5.61 18.39 4 16.4 4H7.6m9.65 1.5a1.25 1.25 0 0 1 1.25 1.25A1.25 1.25 0 0 1 17.25 8 1.25 1.25 0 0 1 16 6.75a1.25 1.25 0 0 1 1.25-1.25M12 7a5 5 0 0 1 5 5 5 5 0 0 1-5 5 5 5 0 0 1-5-5 5 5 0 0 1 5-5m0 2a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3z"
          />
        </svg>
      </a>
      <a
        href="some Pinterest url"
        rel="noreferrer"
        target="_blank"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="PinterestIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M9.04 21.54c.96.29 1.93.46 2.96.46a10 10 0 0 0 10-10A10 10 0 0 0 12 2 10 10 0 0 0 2 12c0 4.25 2.67 7.9 6.44 9.34-.09-.78-.18-2.07 0-2.96l1.15-4.94s-.29-.58-.29-1.5c0-1.38.86-2.41 1.84-2.41.86 0 1.26.63 1.26 1.44 0 .86-.57 2.09-.86 3.27-.17.98.52 1.84 1.52 1.84 1.78 0 3.16-1.9 3.16-4.58 0-2.4-1.72-4.04-4.19-4.04-2.82 0-4.48 2.1-4.48 4.31 0 .86.28 1.73.74 2.3.09.06.09.14.06.29l-.29 1.09c0 .17-.11.23-.28.11-1.28-.56-2.02-2.38-2.02-3.85 0-3.16 2.24-6.03 6.56-6.03 3.44 0 6.12 2.47 6.12 5.75 0 3.44-2.13 6.2-5.18 6.2-.97 0-1.92-.52-2.26-1.13l-.67 2.37c-.23.86-.86 2.01-1.29 2.7v-.03z"
          />
        </svg>
      </a>
    </div>
  </div>
  <div
    className="pages"
  >
    <a
      href="_#"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 page-link-title css-ahj2mt-MuiTypography-root"
      >
        contact us
      </p>
    </a>
    <a
      href="_#"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 page-link-title css-ahj2mt-MuiTypography-root"
      >
        about us
      </p>
    </a>
    <a
      href="_#"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 page-link-title css-ahj2mt-MuiTypography-root"
      >
        subscribe
      </p>
    </a>
  </div>
</div>
`;
